app/assets/javascripts/components/overview/details.jsx

```diff
@@ -89,6 +101,19 @@ const Details = createReactClass({
     return true;
   },
 
+  poll() {
+    if (this.state.course.type === 'Editathon') {
+      return setInterval(() => {
+        if (!this.props.editable) {
+          ServerActions.fetch('course', this.state.course.slug);
```

Comment:
This needs to use `this.props.course` instead of state. (No reason not to use props for the type check too.)

With state, if you edit details, make a change to the course slug (not persisted to the server), then cancel, the state will still have a modified slug and it will start throwing 404 errors.

Reply:
Yeah. Makes sense. We want to encapsulate within `details.jsx` without bothering about the state residing outside of it. Will do.